### DISCUSSION TITLE
Html2pdf

### DISCUSSION THREAD

**Discussion by Manishkparmar80** [Budibase/budibase#1636] [Ideas]
Budibase is not having built-in PDF report functionality so can you guide on how to use below mention html2pdf tool into budibase aplication

https://ekoopmans.github.io/html2pdf.js/

**Comment by shogunpurple** [comment]
@Manishkparmar80 this is not yet possible to directly integrate. It is something that we will support in future but it’s not on our immediate roadmap. 

The only way to do something like this would be to build a separate node API in code that will take the attachment URLs from Budibase, download them and run html2pdf over the files.

Thanks for the suggestion.

**Reply by Manishkparmar80** [reply]
@shogunpurple thanks for clarification.

Alternatively,  what best option presently avaiable into Budibase to achive report requirement into APP developed from budibase.

**Reply by shogunpurple** [reply]
@Manishkparmar80 I'm not sure what you mean. If you would like to save reports/files, you can use the attachments feature. Docs can be found here:

https://docs.budibase.com/data/data-types/attachments

**Reply by Manishkparmar80** [reply]
@shogunpurple  attachment is for uploading information into APP. I am talking about downloading information from APP. E.g. i have developed form and captured input and saved record into database. Now based on saved data, i want generate report. E.g. i have booked sales order and now i want to generate sales invoice report based on save data then how to achieve the same in budibase.
